src/os/windows/library_loader_win.cpp

```diff
@@ -0,0 +1,61 @@
+/*
+ * This file is part of OpenTTD.
+ * OpenTTD is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, version 2.
+ * OpenTTD is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details. You should have received a copy of the GNU General Public License along with OpenTTD. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file library_loader_win.cpp Implementation of the LibraryLoader for Windows */
+
+#include "../../stdafx.h"
+
+#include <windows.h>
+
+#include "../../library_loader.h"
+
+#include "../../safeguards.h"
+
+static std::string GetLoadError()
+{
+	auto error = GetLastError();
+	std::string message;
+	char buffer[512];
+	if (FormatMessageA(FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS, NULL, error,
```

Comment:
I worry that this might end up returning garbage for some locales. The error message might not be in English and the current ANSI codepage for the thread might not be UTF-8 (that might not even be supported on the Windows version the user is running), so this could end up generating mojibake when a non-UTF8 non-English string is treated as UTF-8 further down.

Reply:
This is already existing code in our codebase; I just copied it as the other versions are baked into other flows (like network error handling). So it already stood the test of time.

Further more, the result is only used in Debug statements to allow figuring out why a library failed loading. So I am not too fuzzed about it.